File: app/src/main/java/dev/ragnarok/fenrir/model/database/University.java
package dev.ragnarok.fenrir.model.database;

import android.os.Parcel;
import android.os.Parcelable;


public final class University implements Parcelable {

    public static final Creator<University> CREATOR = new Creator<University>() {
        @Override
        public University createFromParcel(Parcel in) {
            return new University(in);
        }

        @Override
        public University[] newArray(int size) {
            return new University[size];
        }
    };
    private final int id;
    private final String title;

    public University(int id, String title) {
        this.id = id;
        this.title = title;
    }

    private University(Parcel in) {
        id = in.readInt();
        title = in.readString();
    }

    public String getTitle() {
        return title;
    }

    public int getId() {
        return id;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(id);
        parcel.writeString(title);
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/model/database/University.kt
package dev.ragnarok.fenrir.model.database

import android.os.Parcel
import android.os.Parcelable

class University : Parcelable {
    val id: Int
    val title: String?

    constructor(id: Int, title: String?) {
        this.id = id
        this.title = title
    }

    internal constructor(`in`: Parcel) {
        id = `in`.readInt()
        title = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(id)
        parcel.writeString(title)
    }

    companion object CREATOR : Parcelable.Creator<University> {
        override fun createFromParcel(parcel: Parcel): University {
            return University(parcel)
        }

        override fun newArray(size: Int): Array<University?> {
            return arrayOfNulls(size)
        }
    }
}